impl/src/main/java/org/ehcache/impl/internal/concurrent/ConcurrentHashMap.java

```diff
@@ -1,31 +1,45 @@
 /*
- * Copyright Terracotta, Inc.
+ * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
```

Comment:
It would be much safer to pick this up from the 166 repo rather than the OpenJDK one, then the only header will be:
```
/*
 * Written by Doug Lea with assistance from members of JCP JSR-166
 * Expert Group and released to the public domain, as explained at
 * http://creativecommons.org/publicdomain/zero/1.0/
 */
```
Our copyright statement is also still accurate since there are TC changes in the file

Reply:
They are probably the same apart from the header. Do you know how to know which jsr166 version went into the latest Java 9? Current HEAD is complicated to integrate.

Reply:
All good now. I've picked HEAD. Was fine after all.